delta/plugins/elasticsearch/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/deletion/EventMetricsDeletionTaskSuite.scala

```diff
@@ -0,0 +1,60 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.deletion
+
+import akka.http.scaladsl.model.Uri.Query
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.{ElasticSearchBulk, QueryBuilder}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.{ElasticSearchClientSetup, EventMetricsProjection}
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.{Anonymous, Subject}
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ProjectRef
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import ch.epfl.bluebrain.nexus.testkit.{CirceLiteral, TestHelpers}
+import munit.AnyFixture
+
+class EventMetricsDeletionTaskSuite
+    extends BioSuite
+    with ElasticSearchClientSetup.Fixture
+    with CirceLiteral
+    with TestHelpers {
+
+  implicit private val subject: Subject = Anonymous
+
+  override def munitFixtures: Seq[AnyFixture[_]] = List(esClient)
+
+  private lazy val client = esClient()
+
+  test("Delete all entries for a given project") {
+    val prefix          = "test"
+    val index           = EventMetricsProjection.eventMetricsIndex(prefix)
+    val projectToDelete = ProjectRef.unsafe("org", "marked-for-deletion")
+    val anotherProject  = ProjectRef.unsafe("org", "another")
+
+    val task = new EventMetricsDeletionTask(client, prefix)
+
+    val operations = List(
+      ElasticSearchBulk.Index(index, "1", json"""{ "project": "$projectToDelete", "number": 1 }"""),
+      ElasticSearchBulk.Index(index, "2", json"""{ "project": "$anotherProject","number" : 2 }"""),
+      ElasticSearchBulk.Index(index, "3", json"""{ "project": "$projectToDelete", "number" : 3 }"""),
+      ElasticSearchBulk.Index(index, "4", json"""{ "project": "$anotherProject", "number" : 4 }""")
+    )
+
+    def countMetrics(project: ProjectRef) =
+      for {
+        query  <- task.searchByProject(project)
```

Comment:
Using this method in an assertion could lead to a situation where a test should fail and it won't. It's probably better to copy/paste the code rather than having an explicit reference

Reply:
This is an ES query so this is fine, we should rework a bit how we interact with ES but this is out of scope

Reply:
I think copy/paste is much safer but I'll leave it to your judgement